.github/workflows/ci.yml

```diff
@@ -1,8 +1,9 @@
 name: CI
-on: [ push ]
+on: [ push, pull_request ]
 
 jobs:
   lint-and-test:
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
     runs-on: ubuntu-20.04
```

Comment:
```suggestion
    runs-on: ubuntu-22.04
```

Reply:
Thanks, I have changed this, but I actually thought you would want to drive a centralised upgrade action for runners...